felucca/backend/execution_manager.py

```diff
@@ -72,103 +73,84 @@ def save_result(self, task_id, status, stderr, stdout):
-            elif dict[key] is not None:
-                new_command_line_input = new_command_line_input+"-"+key+" "+dict[key]+" "
+        #change the input file path
+        logger = Logger().get()
+        task.output = []
+        task.log = []
+        for task_key in task.arguments:
+            if task_key in ["-f","-s"]:
+                for file_key in task.files:
+                    if file_key == task.arguments[task_key]:
+                        task.arguments[task_key] = task.files[file_key]
+                        break
+           
+            elif task_key in ["-R","-F"]:
+                task.arguments[task_key] = os.path.join(f"/tmp/Felucca/{task.task_id}/",task.arguments[task_key])
+
+                task.log.append(task.arguments[task_key])
                 
-        if 'f' not in dict:
-            new_command_line_input = new_command_line_input+" task.executable_file"
-        
-        task.command_line_input = new_command_line_input
+            elif task_key in ["-j"]:
+                task.arguments[task_key] = os.path.join(f"/tmp/Felucca/{task.task_id}/",task.arguments[task_key])
 
-        task.set_result(output=output,log=log)
-        
-        
-    def copy_to_container(self,src,dst,container):
+                task.output.append(task.arguments[task_key])
+        logger.debug(f"for task({task.task_id}),the task.arguments is {task.arguments}")
+    
+    def copy_to_container(self,task,container):
         
-        """copy executable_file into the container from src to dst
+        """copy executable_file into the container
+            this method will copy the input file from "task.files[key]" at backend to "task.files[key]" ( i.e. the same dst becasue this path is unique for each task) inside the given conatiner
     
         Args:
-            src (str): The absolute path of executable file in the backend 
-            dst (str): The absolute path of executable file in the container 
+            task (Task): task object which conatiner the absolute exe file path currently (i.e. each task.files[key])
             container (Container): the docker container created to run this task 
         
         """
         logger = Logger().get()
-        os.chdir(os.path.dirname(src))
-        srcname = os.path.basename(src)
-
-        tar = tarfile.open(src + '.tar', mode='w')
-        try:
-            tar.add(srcname)
-        except Exception as e:
-            logger.error(f"copy_to_container fails, container:{container.name}, Exception: {e}")
-        finally:
-            tar.close()
-    
-        data = open(src + '.tar', 'rb').read()
-        container_dir = os.path.dirname(dst)
-        container.exec_run("mkdir "+container_dir)
-        container.put_archive(container_dir, data)
         
-        path = src + '.tar'
-        if(os.path.exists(path)):
-            os.remove(path)
+        for key in task.files:
+            src = task.files[key]
```

Comment:
Directly use 
`for filename, path in task.files.items():`
to avoid second accessing.